### ISSUE TITLE
Bug: onDayMouseEnter does not trigger when moving from disabled date to enabled date

### ISSUE THREAD

**Issue by norbertkeresztes** [gpbl/react-day-picker#1613]
### Bug description

The `onDayMouseEnter` event handler does not trigger when the user moves the mouse from a disabled date to a date that is enabled.

### To reproduce

Observe the counter above the calendar which shows how many times the `onDayMouseEnter` event handler triggers.

https://codesandbox.io/s/react-daypicker-forked-vvcnii

#### Steps

1. Navigate to December 2022.
2. Move the mouse over 4th December which is a Sunday and it is a disabled date.
3. Observe the current count after `onDayMouseEnter`.
4. Move the mouse over 5th December without going over any other enabled dates (i.e. directly from 4th to 5th). 5th of December is a Monday and it is an enabled date.
5. Observe that the counter did not change because the `onDayMouseEnter` event handler did not trigger.

### Expected behavior

The `onDayMouseEnter` should always trigger when the user moves the mouse over an enabled date.

[EDIT]
A bit more investigation revealed that it is in fact a React issue. This post explains that there are other issues in React with `onMouseEnter` (and `onMouseLeave`): https://lecstor.com/react-disabled-button-onmouseleave/. The pure HTML version works as expected:

```
<button disabled>Disabled</button><button onmouseenter="javascript:console.log(1)">Enabled</button>
```

Is this something that can be fixed with `react-day-picker`?

**Comment by norbertkeresztes** [comment]
Created a PR to add support for `onPointerEnter` and `onPointerLeave`. These events are correctly triggered by React 17.0.2 and React 18.

**Comment by gpbl** [comment]
Thanks I've merged your fix into v8.4.1
